silver/api/serializers.py

```diff
@@ -465,24 +476,236 @@ def update(self, instance, validated_data):
 
 
 class PaymentProcessorUrl(serializers.HyperlinkedRelatedField):
-    def __init__(self, view_name=None, **kwargs):
-        assert view_name is not None, 'The `view_name` argument is required.'
-        kwargs['read_only'] = True
+    default_validators = [validate_payment_processor]
 
+    def __init__(self, view_name=None, **kwargs):
         super(PaymentProcessorUrl, self).__init__(view_name, **kwargs)
 
-    def to_representation(self, value):
-        kwargs = {'processor_name': value.name}
-        request = self.context.get('request', None)
-        format = self.context.get('format', None)
+    def get_url(self, obj, view_name, request, format):
+        lookup_value = getattr(obj, self.lookup_field)
+        kwargs = {'processor_name': lookup_value}
+        return self.reverse(
+            view_name, kwargs=kwargs, request=request, format=format
+        )
+
+    def get_object(self, view_name, view_args, view_kwargs):
+        return PaymentProcessorManager.get(view_kwargs['processor_name']) or \
+            view_kwargs['processor_name']
 
-        return reverse(self.view_name, kwargs=kwargs, request=request,
-                       format=format)
 
 
 class PaymentProcessorSerializer(serializers.Serializer):
     name = serializers.CharField(max_length=64)
     type = serializers.CharField(max_length=64)
     url = PaymentProcessorUrl(
-        view_name='payment-processor-detail', source='*',
+        view_name='payment-processor-detail', source='*', lookup_field="name",
+        read_only=True
+    )
+
+
+class PaymentMethodUrl(serializers.HyperlinkedRelatedField):
+    def get_url(self, obj, view_name, request, format):
+        kwargs = {'payment_method_id': obj.id,
+                  'customer_pk': obj.customer.id}
```

Comment:
Maybe it doesn't matter in this specific case, but I think `pk` is the getter method meant to always give you the primary key, in case `id` is not the primary key.

http://stackoverflow.com/questions/2165865/django-queries-id-vs-pk